src/cata_tiles.cpp

```diff
@@ -2179,7 +2182,8 @@ bool cata_tiles::draw_terrain( const tripoint &p, const lit_level ll, int &heigh
             // do something to get other terrain orientation values
         }
         const std::string &tname = t.id().str();
-        if( g->m.check_seen_cache( p ) ) {
+        // re-memorize seen terrain in case new connections have been formed
```

Comment:
I'm very concerned that this defeats the caching entirely, and possibly pathologically, which is incidentally why I haven't taken a stab at it myself, I can't see a way to fix it without breaking the cache.

The whole point of the cache is to not call memorize_tile(), but has_terrain_memory_at() is nearly as expensive, and is very frequently going to return true, meaning we call memorize_tile() anyway.

I might have a fix though, we can bypass the cache when ``if( t.obj().connects( connect_group ) ) {`` a few lines up fires.  that pessimizes for any connecting tiles, but those should be relatively rare, and gets us re-memorization.

However, that does mean that the final memorized tile is going to be the "last one seen", so it's still not going to be always the most correct, that would require us to have an ordering of which tiles are preferred somehow so it doesn't "downgrade".